types/react/index.d.ts

```diff
@@ -356,7 +356,7 @@ declare namespace React {
         displayName?: string;
     }
 
-    interface ComponentClass<P = {}, S = ComponentState> extends StaticLifecycle<P, S> {
+    interface ComponentClass<P = {}, S = {}> extends StaticLifecycle<P, any> {
```

Comment:
I'm not fully familiar with the React definitions, but at first glance this seems wrong as `StaticLifecycle` will no longer get a useful `S` for use with `getDerivedStateFromProps`. Judging from @DanielRosenwasser's comment, the previous commit (where `ComponentState` was typed as `any`), might be better.

Reply:
Sounds fair enough. I changed it back to my previous commit